diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index e962733..0edcdcd 100644
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -181,7 +181,7 @@ static u32 force_fps;
 #define H265_DEBUG_IGNORE_CONFORMANCE_WINDOW	0x2000000
 #define H265_DEBUG_NOWAIT_DECODE_DONE_WHEN_STOP   0x4000000
 
-#define MAX_BUF_NUM 24
+#define MAX_BUF_NUM 16
 #define MAX_REF_PIC_NUM 24
 #define MAX_REF_ACTIVE  16
 
@@ -215,7 +215,7 @@ static u32 dynamic_buf_num_margin;
 #else
 static u32 buf_alloc_width;
 static u32 buf_alloc_height;
-static u32 dynamic_buf_num_margin = 7;
+static u32 dynamic_buf_num_margin = 16;
 #endif
 static u32 buf_alloc_size;
 static u32 re_config_pic_flag;
@@ -236,7 +236,7 @@ bit[3]: 1, if blackout is not 1, do not release current
 /* set to 1 for fast play;
 	set to 8 for other case of "keep last frame"
 */
-static u32 buffer_mode = 1;
+static u32 buffer_mode = 5;
 /* buffer_mode_dbg: debug only*/
 static u32 buffer_mode_dbg = 0xffff0000;
 /**/
diff --git a/drivers/amlogic/amports/vvp9.c b/drivers/amlogic/amports/vvp9.c
index d129327..6c4c868 100644
--- a/drivers/amlogic/amports/vvp9.c
+++ b/drivers/amlogic/amports/vvp9.c
@@ -75,7 +75,7 @@
 #define VP9_EOS                        0xf2
 #define HEVC_ACTION_DONE                0xff
 
-#define MAX_BUF_NUM 24
+#define MAX_BUF_NUM 16
 #define MAX_REF_ACTIVE  16
 #define VF_POOL_SIZE        32
 
@@ -1691,7 +1691,7 @@ static u32 dynamic_buf_num_margin;
 #else
 static u32 buf_alloc_width;
 static u32 buf_alloc_height;
-static u32 dynamic_buf_num_margin = 7;
+static u32 dynamic_buf_num_margin = 16;
 #endif
 static u32 buf_alloc_depth = 10;
 static u32 buf_alloc_size;
@@ -1712,7 +1712,7 @@ bit[3]: 1, if blackout is not 1, do not release current
 /* set to 1 for fast play;
 	set to 8 for other case of "keep last frame"
 */
-static u32 buffer_mode = 1;
+static u32 buffer_mode = 5;
 /* buffer_mode_dbg: debug only*/
 static u32 buffer_mode_dbg = 0xffff0000;
 /**/
