diff --git a/xbmc/windowing/gbm/DRMUtils.cpp b/xbmc/windowing/gbm/DRMUtils.cpp
index bc5e7d1..0c99d14 100644
--- a/xbmc/windowing/gbm/DRMUtils.cpp
+++ b/xbmc/windowing/gbm/DRMUtils.cpp
@@ -203,7 +203,7 @@ bool CDRMUtils::GetPreferredMode()
 bool CDRMUtils::InitDrm(drm *drm)
 {
   m_drm = drm;
-  const char *device = "/dev/dri/card0";
+  const char *device = "/dev/dri/card1";
 
   m_drm->fd = open(device, O_RDWR);
 

